### PR TITLE
ED: Add inputs for the off-diagonal terms of the platform moment of inertia matrix

### PR DESCRIPTION
**Pull request by luwang00**
**Feature or improvement description**
This PR adds three new inputs in ElastoDyn for the off-diagonal terms of the platform moment of inertia matrix, Ixy, Iyz, and Izx. The ElastoDyn platform equations of motion are also updated to include these terms. The ElastoDyn input documentation is updated as well.

**Impacted areas of the software**
ElastoDyn

**Test results, if applicable**
r-test ElastoDyn input files have been updated with the three new inputs. The off-diagonal moment of inertia terms are set to zero in all existing r-tests, so no change to the results.

### CODE REVIEW

File: modules/elastodyn/src/ElastoDyn_Registry.txt [hunk 1]
```diff
@@ -153,6 +153,9 @@ typedef	^	ED_InputFile	ReKi	PtfmMass	-	-	-	"Platform mass"	kg
 typedef	^	ED_InputFile	ReKi	PtfmRIner	-	-	-	"Platform inertia for roll tilt rotation about the platform CM"	"kg m^2"
 typedef	^	ED_InputFile	ReKi	PtfmPIner	-	-	-	"Platform inertia for pitch tilt rotation about the platform CM"	"kg m^2"
 typedef	^	ED_InputFile	ReKi	PtfmYIner	-	-	-	"Platform inertia for yaw rotation about the platform CM"	"kg m^2"
+typedef	^	ED_InputFile	ReKi	PtfmXYIner	-	-	-	"Platform xy inertia about the platform CM"	"kg m^2"
+typedef	^	ED_InputFile	ReKi	PtfmYZIner	-	-	-	"Platform yz inertia for pitch tilt rotation about the platform CM"	"kg m^2"
```
Comment (jjonkman) [inline modules/elastodyn/src/ElastoDyn_Registry.txt:157]: Minor comment: Delete "for pitch tilt rotation".

File: modules/elastodyn/src/ElastoDyn_Registry.txt [hunk 2]
```diff
@@ -672,6 +675,9 @@ typedef	^	ParameterType	ReKi	PtfmMass	-	-	-	"Platform mass"
 typedef	^	ParameterType	ReKi	PtfmPIner	-	-	-	"Platform inertia for pitch tilt rotation about the platform CM."
 typedef	^	ParameterType	ReKi	PtfmRIner	-	-	-	"Platform inertia for roll tilt rotation about the platform CM."
 typedef	^	ParameterType	ReKi	PtfmYIner	-	-	-	"Platform inertia for yaw rotation about the platform CM."
+typedef	^	ParameterType	ReKi	PtfmXYIner	-	-	-	"Platform xy inertia about the platform CM"	"kg m^2"
+typedef	^	ParameterType	ReKi	PtfmYZIner	-	-	-	"Platform yz inertia for pitch tilt rotation about the platform CM"	"kg m^2"
```
Comment (jjonkman) [inline modules/elastodyn/src/ElastoDyn_Registry.txt:679]: Minor comment: Delete "for pitch tilt rotation".

File: modules/elastodyn/src/ElastoDyn_Registry.txt [hunk 3]
```diff
@@ -672,6 +675,9 @@ typedef	^	ParameterType	ReKi	PtfmMass	-	-	-	"Platform mass"
...
+typedef	^	ParameterType	ReKi	PtfmXZIner	-	-	-	"Platform xz inertia for yaw rotation about the platform CM"	"kg m^2"
```
Comment (jjonkman) [inline modules/elastodyn/src/ElastoDyn_Registry.txt:680]: Minor comment: Delete "for yaw rotation".

**Review by jjonkman — COMMENTED** [review]
Thanks @luwang00.  I agree with these equations, including the signs in the inertia dyadic (given that you've defined the cross-inertias with the negative sign).  Just a few minor comments on registry descriptions...

File: modules/elastodyn/src/ElastoDyn_Registry.txt (hunk 3 above)
Comment (luwang00) [reply modules/elastodyn/src/ElastoDyn_Registry.txt]: @jjonkman Thanks for reviewing the changes! The registry descriptions for the off-diagonal moment of inertia terms are now corrected.

**Comment by luwang00** [comment]
Hi @andrew-platt, I think this PR is ready to merge.

**Review by andrew-platt — APPROVED** [review]

**Pull request merged**